tests/Shared/nuget-with-package-source-mapping.config

```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
```

Comment:
This is the 3rd time we have a nuget.config copied in the repo. Is there a way to use the root and just tweak it in places we need it to be special?

Reply:
Yeah, we had a task doing that earlier because we needed to put in the path to the built nugets. But now we use `%BUILT_NUGETS_PATH%` as the source in the nuget.config for tests, so it always resolves it from the environment variable.

I'll bring the task back to get the sources from the root nuget.config, add the source for built-nugets, and add the package source mapping as needed.
Added this as follow up work - https://github.com/dotnet/aspire/issues/4946 .